core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/AddOperationsToChain.java

```diff
@@ -0,0 +1,154 @@
+import uk.gov.gchq.gaffer.store.operationdeclaration.OperationDeclarations;
+import uk.gov.gchq.gaffer.user.User;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+public class AddOperationsToChain implements GraphHook {
+    private static final JSONSerialiser JSON_SERIALISER = new JSONSerialiser();
+
+    private List<Operation> start;
+    private List<Operation> end;
+    private Map<String, List<Operation>> before;
+    private Map<String, List<Operation>> after;
+
+    public AddOperationsToChain() {
+    }
+
+    public AddOperationsToChain(final String addOperationsPath) throws IOException {
+
+        Path path = Paths.get(addOperationsPath);
+        AddOperationsToChain addOperations;
+        if (path.toFile().exists()) {
+            addOperations = fromJson(Files.readAllBytes(path));
+        } else {
+            addOperations = fromJson(StreamUtil.openStream(OperationDeclarations.class, addOperationsPath));
+        }
+        if (addOperations.getStart() != null) {
+            this.setStart(addOperations.getStart());
+        }
+        if (addOperations.getEnd() != null) {
+            this.setEnd(addOperations.getEnd());
+        }
+        if (addOperations.getBefore() != null) {
+            this.setBefore(addOperations.getBefore());
+        }
+        if (addOperations.getAfter() != null) {
+            this.setAfter(addOperations.getAfter());
+        }
+    }
+
+    public List<Operation> getStart() {
+        return start;
+    }
+
+    public void setStart(final List<Operation> start) {
+        this.start = start;
+    }
+
+    public List<Operation> getEnd() {
+        return end;
+    }
+
+    public void setEnd(final List<Operation> end) {
+        this.end = end;
+    }
+
+    public Map<String, List<Operation>> getBefore() {
+        return before;
+    }
+
+    public void setBefore(final Map<String, List<Operation>> before) {
+        this.before = before;
+    }
+
+    public Map<String, List<Operation>> getAfter() {
+        return after;
+    }
+
+    public void setAfter(final Map<String, List<Operation>> after) {
+        this.after = after;
+    }
+
+    @Override
+    public void preExecute(final OperationChain<?> opChain, final User user) {
+        OperationChain<?> newOpChain = new OperationChain<>();
+
+        if (start != null) {
+            newOpChain.getOperations().addAll(start);
+        }
+        if (opChain != null) {
+            for (final Operation originalOp : opChain.getOperations()) {
+
+                for (final String beforeOpKey : before.keySet()) {
+                    if (originalOp.getClass().getName().contains(beforeOpKey)) {
+                        newOpChain.getOperations().addAll(before.get(beforeOpKey));
+                    }
+                    newOpChain.getOperations().add(originalOp);
```

Comment:
If before is empty then the originalOp would never be added. If before contains multiple operations then the original operation is added multiple times.